### ISSUE TITLE
@typescript-eslint/restrict-plus-operands causes error "ESLint encountered a parsing error"

### ISSUE THREAD

**Issue by NaderNashed** [typescript-eslint/typescript-eslint#3882]
Adding @typescript-eslint/restrict-plus-operands, then after setting the parseroptions.project to my tsconfig I get the error  "ESLint encountered a parsing error"

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```json
{
  "parserOptions": { "project": "./tsconfig.json" },
  "parser": "@typescript-eslint/parser",
  "plugins": [ "@typescript-eslint" ],
  "rules": { "@typescript-eslint/restrict-plus-operands": "error" }
}
```

```TS
   const foo = '5.5' + 5;
```

```json
{
  "compilerOptions": {
    "module": "commonjs",
    "target": "es6",
    "lib": ["es6"],
    "sourceMap": true
  },
  "exclude": [
    "node_modules"
  ]
}
```

**Expected Result**

For the rule to work and shows an error about the line of code above.

What did you expect to happen?
Please be specific here - list the exact lines and messages you expect.

**Actual Result**

Instead I got the error "ESLint encountered a parsing error"

**Additional Info** 

Did eslint throw an exception?
No
Please run your lint again with the --debug flag, and dump the output below.
i.e. eslint --ext ".ts,.js" src --debug

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` |4.31.1 |
| `@typescript-eslint/parser`        | 4.31.1 |
| `TypeScript`                       | 4.0.5 |
| `ESLint`                           | 7.32.0 |
| `node`                             | 14.14.7 |

**Comment by bradzacher** [comment]
> ESLint encountered a parsing error

Is that it? That's all that ESLint told you?
Could you please provide more information like an isolated repro?
Or as suggested the `--debug` logs?

**Comment by NaderNashed** [comment]
I found out the rules for as expected from the command line. EsLint encountered a parsing error when I was loading eslint in visual studio 2019

**Comment by NaderNashed** [comment]
So this is an integration issue. Do you consider an issue on your side of on Visual studio side? Note that it started only when I set parserOptions.porject.

**Comment by NaderNashed** [comment]
[repro.zip](https://github.com/typescript-eslint/typescript-eslint/files/7157815/repro.zip)
Here is the minimal repro.

**Comment by bradzacher** [comment]
Given that it works from the CLI but not from VS 2019 it's either VS issue or you've got a config problem.

Given you just use a relative path - I'd wager it's the latter.

Our getting started guides tell you how to configure it correctly so it works consistently: https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/TYPED_LINTING.md

**Comment by NaderNashed** [comment]
Thanks for the last link. Now working in IDE

**Comment by NaderNashed** [comment]
Thanks for the nice work, This rule is very useful.

I just  have a suggestion that could save everybody's time. As I am new to find an individual rule which happened  to require type information. All I knew about it was its name and by googling it I landed on its page in Github. https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/restrict-plus-operands.md
Unfortunately no settings was there { "@typescript-eslint/restrict-plus-operands": "error" }. If the page had the complete settings same as in the link @bradzacher sent to me or at least there was a link to it, that would have spared me a half day of research and spared the time @bradzacher spent replying to me.

**Comment by bradzacher** [comment]
There are a large, large number of rules - including complete configuration for every single one of them including all required parserOptions or environment config is overly verbose, repetitive and hard to maintain due to the duplication.

Instead we centralise docs on how to do the broad, one-off configuration in our getting started guide (https://github.com/typescript-eslint/typescript-eslint/tree/master/docs/getting-started/linting) which runs through all of the configuration.
The getting started guide is linked to from the root readmes for user-visible packages.
- https://github.com/typescript-eslint/typescript-eslint#getting-started--installation
- https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin#getting-started
- https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/parser#getting-started
- https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/scope-manager#getting-started
- https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/typescript-estree#getting-started

These root readmes are all shown on the NPM page for the packages.

The issue is that by googling for the rule, leading you to hop into the middle of the setup process, and thus got lost there.
